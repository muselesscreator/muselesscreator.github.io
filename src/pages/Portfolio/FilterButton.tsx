import classNames from 'classnames';

type FilterButtonProps = {
  isActive: boolean;
  filter: string;
  handleFilterChange: (category: string) => () => void;
};
const FilterButton = ({
  isActive,
  filter,
  handleFilterChange,
}: FilterButtonProps) => {
  return (
    <li className="filter-item">
      <button
        className={classNames({ active: isActive })}
        onClick={handleFilterChange(filter)}
      >
        {filter}
      </button>
    </li>
  );
};
export default FilterButton;
