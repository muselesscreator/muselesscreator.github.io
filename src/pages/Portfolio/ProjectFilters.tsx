import Select from '~/components/Select';
import { StrictDict } from '@muselesscreator/strict-dict';

import FilterButton from './FilterButton';
import { ProjectCategories } from './data';

type ProjectFiltersProps = {
  projectFilter: string;
  handleCategoryChange: (category: string) => () => void;
  handleFilterChange: (value: unknown) => void;
};

export const filterOptions = [
  { label: ProjectCategories.all, value: ProjectCategories.all },
  { label: ProjectCategories.public, value: ProjectCategories.public },
  { label: ProjectCategories.openSource, value: ProjectCategories.openSource },
];

export const categories = [
  ProjectCategories.all,
  ProjectCategories.public,
  ProjectCategories.openSource,
];

export const classNames = StrictDict({
  list: 'filter-list',
  selectBox: 'filter-select-box',
}) as Record<string, string>;

const ProjectFilters = ({
  projectFilter,
  handleCategoryChange,
  handleFilterChange,
}: ProjectFiltersProps) => {
  return (
    <div>
      <ul className={classNames.list}>
        {categories.map((category) => (
          <FilterButton
            key={category}
            isActive={projectFilter === category}
            handleFilterChange={handleCategoryChange}
            filter={category}
          />
        ))}
      </ul>
      <div className={classNames.selectBox}>
        <Select
          label="Select category"
          options={filterOptions}
          value={projectFilter}
          handleChange={handleFilterChange}
        />
      </div>
    </div>
  );
};

export default ProjectFilters;
