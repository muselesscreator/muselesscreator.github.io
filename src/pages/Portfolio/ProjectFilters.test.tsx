import {
  vi,
  describe,
  test,
  expect,
} from 'vitest';
import { shallow, setMockName } from '@muselesscreator/react-shallow-snapshot';
import FilterButton from './FilterButton';
import ProjectFilters, {
  filterOptions,
  categories,
} from './ProjectFilters';

vi.mock('./FilterButton', () => ({ default: 'FilterButton' }));
vi.mock('~/components/Select', () => ({ default: 'Select' }));

const props = {
  projectFilter: categories[0],
  handleCategoryChange: vi.fn(),
  handleFilterChange: vi.fn(),
};
setMockName(props.handleCategoryChange, 'props.handleCategoryChange');
setMockName(props.handleFilterChange, 'props.handleFilterChange');

describe('ProjetFilters component', () => {
  test('snapshot', () => {
    const el = shallow(<ProjectFilters {...props} />);
    expect(el.snapshot).toMatchSnapshot();
    const filterButtons = el.instance.findByType(FilterButton);
    const select = el.instance.findByType('Select')[0];
    expect(filterButtons).toHaveLength(filterOptions.length);
    expect(filterButtons[0].props).toEqual({
      isActive: true,
      handleFilterChange: props.handleCategoryChange,
      filter: categories[0],
    });
    expect(filterButtons[1].props).toEqual({
      isActive: false,
      handleFilterChange: props.handleCategoryChange,
      filter: categories[1],
    });
    expect(filterButtons[2].props).toEqual({
      isActive: false,
      handleFilterChange: props.handleCategoryChange,
      filter: categories[2],
    });
    expect(select.props).toEqual({
      label: 'Select category',
      options: filterOptions,
      value: props.projectFilter,
      handleChange: props.handleFilterChange,
    });
  });
});
