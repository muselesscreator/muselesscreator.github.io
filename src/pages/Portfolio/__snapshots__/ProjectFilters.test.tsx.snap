// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjetFilters component > snapshot 1`] = `
"<div>
  <ul
    className="filter-list"
  >
    <FilterButton
      filter="All"
      handleFilterChange={[Function props.handleCategoryChange]}
      isActive={true}
    />
    <FilterButton
      filter="Public Packages"
      handleFilterChange={[Function props.handleCategoryChange]}
      isActive={false}
    />
    <FilterButton
      filter="Open Source Contributions"
      handleFilterChange={[Function props.handleCategoryChange]}
      isActive={false}
    />
  </ul>
  <div
    className="filter-select-box"
  >
    <Select
      handleChange={[Function props.handleFilterChange]}
      label="Select category"
      options={
        Array [
          Object {
            "label": "All",
            "value": "All",
          },
          Object {
            "label": "Public Packages",
            "value": "Public Packages",
          },
          Object {
            "label": "Open Source Contributions",
            "value": "Open Source Contributions",
          },
        ]
      }
      value="All"
    />
  </div>
</div>"
`;
